File: C++/mirror.cpp
#include<iostream>

struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 };

class Solution {
public:
    TreeNode* Mirror(TreeNode* pRoot) {
        if (!pRoot) {
            return nullptr; 
        }
        TreeNode* tmp = new TreeNode(0); 
        tmp=pRoot->left;
        pRoot->left=pRoot->right;
        pRoot->right=tmp;
        Mirror(pRoot->right);
        Mirror(pRoot->left);    
        return pRoot;
    }
};

void printTree(TreeNode* root) {
    if (root) {
        std::cout << root->val << " ";
        printTree(root->left);
        printTree(root->right);
    }
}

void deleteTree(TreeNode* root) {
    if (root) {
        deleteTree(root->left);
        deleteTree(root->right);
        delete root;
    }
}

int main(){
    TreeNode* pRoot=new TreeNode(8);
    pRoot->left=new TreeNode(6);
    pRoot->right=new TreeNode(10);
    pRoot->left->left=new TreeNode(5);
    pRoot->left->right=new TreeNode(7);
    pRoot->right->left=new TreeNode(9);
    pRoot->right->right=new TreeNode(11);
    Solution solution;
    TreeNode* root=solution.Mirror(pRoot);

    std::cout<<"Mirror tree: ";
    printTree(root);
    std::cout << std::endl;

    deleteTree(root);

    return 0;
}

File: python/Mirror.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
        
class Solution:
    def Mirror(self , pRoot: TreeNode) -> TreeNode:
        if not pRoot:
            return None 
        tmp=TreeNode(0)
        tmp = pRoot.left
        pRoot.left=pRoot.right
        pRoot.right=tmp  
        self.Mirror(pRoot.left)
        self.Mirror(pRoot.right)    
        return pRoot
 
    def printTree(self,root:TreeNode):
        if root is None:
            return 
        print(root.val,end=' ')
        self.printTree(root.left)
        self.printTree(root.right)
    
root=TreeNode(8)
root.left=TreeNode(6)
root.right=TreeNode(10)
root.left.left=TreeNode(5)
root.left.right=TreeNode(7)
root.right.left=TreeNode(9)
root.right.right=TreeNode(11)
solution=Solution()
mirror_root=solution.Mirror(root)
solution.printTree(mirror_root)
print()
